src/test/java/org/opensearch/knn/index/query/KNNQueryBuilderTests.java

```diff
@@ -1066,11 +1066,18 @@ public void testDoRewrite_whenFilterSet_thenSuccessful() {
             .filter(rewrittenFilter)
             .k(K)
             .build();
+
+        QueryBuilder filterBefore = filter;
+
         // When
         KNNQueryBuilder knnQueryBuilder = KNNQueryBuilder.builder().fieldName(FIELD_NAME).vector(QUERY_VECTOR).filter(filter).k(K).build();
 
         QueryBuilder actual = knnQueryBuilder.rewrite(context);
 
+        QueryBuilder filterAfter = knnQueryBuilder.getFilter();
+
+        assertEquals(filterBefore, filterAfter);
```

Comment:
nit: We don't want to just check filter here, we want to make sure no field is manipulated we can have a stronger check here. this makes sure none of the fields are manipulated unintentionally

```suggestion
        assertEquals(knnQueryBuilder, KNNQueryBuilder.builder().fieldName(FIELD_NAME).vector(QUERY_VECTOR).filter(filter).k(K).build());
```